### DISCUSSION TITLE
Looks like `redirect().back()` isn't working as it should?

### DISCUSSION THREAD

**Discussion by sooluh** [adonisjs/core#3816] [Help] [answered]
Hello AdonisJS maintainers.

I'm new to trying to use AdonisJS in my production. I saw some code from websites built with this framework using the `redirect().back()` function without hesitation.

But after I tried it myself, I actually get a redirect to the main route of the website, after I understand some code from AdonisJS, it turns out that this `redirect().back()` uses a referrer header.

I try to inspect on my website, and the value of the referrer header I get is correct, the main route of the website. Maybe the `redirect().back()` function doesn't have an error, but am I missing a configuration that causes all referrer header on my website to show the main route?

Please enlighten, apologize, and thank you.

**Comment by Barbapapazes** [comment]
Hello,

Can you provide a reproduction repo to view this error? Thanks

**Reply by sooluh** [reply]
After I tried to run some open source code that uses AdonisJS, the results are the same, maybe because I use WSL to run it?

**Reply by Barbapapazes** [reply]
When you say the main route, you mean `/` ?

**Reply by sooluh** [reply]
Precisely

**Reply by Barbapapazes** [reply]
And can you describe your use case? When and where did you use the method?

**Reply by sooluh** [reply]
I call it in the `process` method in the `auth` controller and it is sent from the `index` method on the same controller.

I checked on the network there it says that the referrer header brought is the main route :)

**Reply by Barbapapazes** [reply]
Hello,

I [create a repo using a working redirect back](https://github.com/Barbapapazes/reproduction-redirect-back) and some of the name given here. Can you look at it?

When you navigate to `/auth/login`, you can then submit your form and the redirect back which is in `/auth/process` redirect the request to `/auth/login`! (and 'process' is print in the console).

I use 'web' to [start this new project](https://docs.adonisjs.com/guides/installation#project-structure)!

**Reply by sooluh** [reply]
I can't run it because you forgot the `.adonisrc.json` file to commit.

**Reply by Barbapapazes** [reply]
I fixe it! Sorry

**Reply by sooluh** [reply]
Hmmm, yes it works! Looks like I have a misconfiguration huh :) But I can't find it :(

**Comment by sooluh** [comment, marked as answer]
Well, after a long time of analyzing, finally found the main problem.

I'm using WSL and the referrer header doesn't represent the real thing at all (it can only show the root domain, not the full url). Once I tried running it on Windows (main host), it worked!

On the server, because of Nginx configuration, I set "strict-origin" in the header "Referrer-Policy" before, after I commented (disabled) the configuration line, everything worked normally.

**Reply by thetutlage** [reply]
Thanks for sharing the solution. I am sure it will help others in the future
